server/internal/storage/storage.go

```diff
@@ -34,3 +36,42 @@ func New(ctx context.Context, uri, db string) (*Storage, error) {
 func (s Storage) Disconnect(ctx context.Context) error {
 	return s.Client.Disconnect(ctx)
 }
+
+func (s *Storage) GetPlantsWithCareRules(ctx context.Context) ([]*models.Plant, error) {
+	collection := s.DataBase.Collection("plants")
+	filter := bson.M{"care_rules": bson.M{"ne": nil}}
```

Comment:
```suggestion
	filter := bson.M{"care_rules": bson.M{"$ne": nil}}
```
будто бы без $ оператор не должен правильно сработать